lib/jekyll/commands/serve.rb

```diff
@@ -33,7 +33,9 @@ def init_with_program(prog)
             cmd.action do |_, opts|
               opts["serving"] = true
               opts["watch"  ] = true unless opts.key?("watch")
+              cfg = opts["config"]
               Build.process(opts)
+              opts["config"] = cfg
```

Comment:
Thanks for the review! As i said in my initial comment the config option gets lost in the build process on this specific [line](https://github.com/jekyll/jekyll/blob/632f3fd8de0c63813f46240216cb12905d829968/lib/jekyll/configuration.rb#L146) and the serve calls, which is started after the build process has finished has no information about the given configs anymore. I tried to replace the `delete` call with `fetch` but this was causing other tests to fail so i took another approach.